src/api_server/src/lib.rs

```diff
@@ -297,6 +298,13 @@ impl ApiServer {
             _ => None,
         };
 
+        if let VmmAction::SetMmdsConfiguration(ref config) = *vmm_action {
+            let mmds_response = self.set_mmds_config(*config);
+            if mmds_response.status() != StatusCode::NoContent {
+                return mmds_response;
+            }
+        }
+
```

Comment:
Is this truly a `VmmAction` or not? The extra logic here is just for logging purposes. `VmmAction`s are supposed to run in the VMM. What is the purpose of calling `self.set_mmds_config()` here if we call it in the VMM as well? We should have a clear separation here. If we can perform all the necessary logic here, we should not have anything extra in the VMM. If we can't, we should move all of it there (if it pertains to the VMM).

Reply:
The reason behind this is actually the separation we have between MMDS network stack and MMDS data store itself. We have to split this setter into two, if we want to be able to use `/mmds/config` to set both the version and also the IPv4 address.

Firstly, we need to call `self.set_mmds_config()` in the api_server (here) because the MMDS data store is accessible from only from within `ApiServer` - so it's the only place we could access the `MMDS` data store and set its version.

Secondly, the MMDS network configuration is accessible from the `VmResources` (inside the `PrebootApiController`), so in order to be able to pass the IPv4 address, we need to keep the setter there, too.